sklearn/tests/test_calibration.py

```diff
@@ -618,3 +621,169 @@ def test_calibration_votingclassifier():
     calib_clf = CalibratedClassifierCV(base_estimator=vote, cv="prefit")
     # smoke test: should not raise an error
     calib_clf.fit(X, y)
+
+
+@pytest.fixture(scope="module")
+def iris_data():
+    return load_iris(return_X_y=True)
+
+
+@pytest.fixture(scope="module")
+def iris_data_binary(iris_data):
+    X, y = iris_data
+    return X[y < 2], y[y < 2]
+
+
+def test_calibration_display_validation(pyplot, iris_data, iris_data_binary):
+    X, y = iris_data
+    X_binary, y_binary = iris_data_binary
+
+    reg = LinearRegression().fit(X, y)
+    msg = "'estimator' should be a fitted classifier"
+    with pytest.raises(ValueError, match=msg):
+        CalibrationDisplay.from_estimator(reg, X, y)
+
+    clf = LinearSVC().fit(X, y)
+    msg = "response method predict_proba is not defined in"
+    with pytest.raises(ValueError, match=msg):
+        CalibrationDisplay.from_estimator(clf, X, y)
+
+    clf = LogisticRegression()
+    with pytest.raises(NotFittedError):
+        CalibrationDisplay.from_estimator(clf, X, y)
+
+
+@pytest.mark.parametrize("constructor_name", ["from_estimator", "from_predictions"])
+def test_calibration_display_non_binary(pyplot, iris_data, constructor_name):
+    X, y = iris_data
+    clf = DecisionTreeClassifier()
+    clf.fit(X, y)
+    y_prob = clf.predict_proba(X)
+
+    constructor = getattr(CalibrationDisplay, constructor_name)
+    params = (clf, X, y) if constructor_name == "from_estimator" else (y, y_prob)
+
+    msg = (
+        "Only binary classification is supported."
+        if constructor_name == "from_estimator"
+        else "y should be a 1d array, got an array of shape"
+    )
+    with pytest.raises(ValueError, match=msg):
+        constructor(*params)
```

Comment:
This test is failing due to the error message:

```
        clf.fit(X, y)
        y_prob = clf.predict_proba(X)
    
        constructor = getattr(CalibrationDisplay, constructor_name)
        params = (clf, X, y) if constructor_name == "from_estimator" else (y, y_prob)
    
        msg = (
            "Only binary classification is supported."
            if constructor_name == "from_estimator"
            else "y should be a 1d array, got an array of shape"
        )
        with pytest.raises(ValueError, match=msg):
>           constructor(*params)
E           AssertionError: Regex pattern 'Only binary classification is supported.' does not match 'DecisionTreeClassifier should be a binary classifier'.
```

I think that here, it could be more readable to make the if/else explicit:


```python
if constructor_name == "from_estimator":
    err_msg = "xxx"
    with pytest.raises(ValueError, match=err_msg):
        CalibrationDisplay.from_estimator(clf, X, y)
else:
    err_msg = "yyy"
    with pytest.raises(ValueError, match=err_msg):
        CalibrationDisplay.from_predictions(y, y_prob)
```